app/src/main/java/com/onboarding/calculator/MainActivity.java

```diff
@@ -4,11 +4,42 @@
 
 import android.os.Bundle;
 
+import com.onboarding.calculator.databinding.ActivityMainBinding;
+import com.onboarding.calculator.mvp.model.CalculatorModel;
+import com.onboarding.calculator.mvp.presenter.CalculatorPresenter;
+import com.onboarding.calculator.mvp.view.CalculatorView;
+
 public class MainActivity extends AppCompatActivity {
 
+    private ActivityMainBinding binding;
+    private CalculatorPresenter presenter;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_main);
+        binding = ActivityMainBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+        presenter = new CalculatorPresenter(new CalculatorModel(), new CalculatorView(this, binding));
```

Comment:
as a minor improve you can create a contract for presenter, model and view

Reply:
Solved!